index.js

```diff
@@ -23,19 +24,27 @@ module.exports = function(options) {
   var replaceString = options.replaceString || /^gulp(-|\.)/;
   var camelizePluginName = options.camelize === false ? false : true;
   var lazy = 'lazy' in options ? !!options.lazy : true;
-  var requireFn = options.requireFn || require;
   var renameObj = options.rename || {};
 
-  if (typeof config === 'string') {
-    config = require(config);
+  if(typeof options.requireFn === 'function') {
+    requireFn = options.requireFn;
+  } else if(typeof config === 'string') {
+    requireFn = function (name) {
+      var searchFor = path.join('node_modules', name);
+      return require(findup(searchFor, {cwd: path.dirname(config)}));
+    };
```

Comment:
could we add a comment around this block explaining what it's there for? Maybe with a link to this PR or the issue.